### PR TITLE
New Entry and Terms: Python modules

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

From the Target Keyword list. Gives a location for all the 3rd party Python modules we don't need to go into detail about.
 
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Finished my review and left you some comments/suggestions. My gut-feeling is that the term entries for Numpy and Pandas seem redundant in content but I understand why they're here. For now, they're okay to stay.

**Comment by SSwiniarski** [comment]
@Dusch4593 Do you think that the Numpy and Pandas references would be better linked to in the module article than as terms? The more I think about it, the more I think there's a potential for confusion, especially when linking out to the Docs entry.

**Comment by SSwiniarski** [comment]
@Dusch4593 I completed all your suggestions.

**Comment by SSwiniarski** [comment]
@Dusch4593 anything more you think needs to be done?

File: content/python/concepts/modules/modules.md
Comment (Dusch4593) [inline content/python/concepts/modules/modules.md:78]:
```suggestion
- [NumPy](https://www.codecademy.com/resources/docs/numpy): a popular open source Python library used for complex mathematical operations and multi-dimensional [arrays](https://www.codecademy.com/resources/docs/general/array).
```

**Review by Dusch4593 — APPROVED** [review]
@SSwiniarski This LGTM! 👍🏻 Just left one last suggestion, let's remove the link to the General: Open Source entry since this link will break after #935 is merged and the new topic is created. I can make this change on that PR.

File: content/python/concepts/modules/modules.md
Comment (yangc95) [inline content/python/concepts/modules/modules.md:62]:
```suggestion
Python comes with several different built-in modules that provide a variety of functions. They include:
```

File: content/python/concepts/modules/modules.md
Comment (yangc95) [inline content/python/concepts/modules/modules.md:76]:
```suggestion
These are third-party Python modules that have topic entries:
```

File: content/python/concepts/modules/modules.md
Comment (yangc95) [inline content/python/concepts/modules/modules.md:79]:
```suggestion
- [NumPy](https://www.codecademy.com/resources/docs/numpy): a popular open-source Python library used for complex mathematical operations and multi-dimensional [arrays](https://www.codecademy.com/resources/docs/general/array).
- [Pandas](https://www.codecademy.com/resources/docs/pandas): a popular open-source Python module used for data analysis and manipulation.
```

Comment (yangc95) [reply content/python/concepts/modules/modules.md:79]: Docs entries use "open-source" more so than "open source"

File: content/python/concepts/modules/terms/pytorch/pytorch.md
Comment (yangc95) [inline content/python/concepts/modules/terms/pytorch/pytorch.md:3]:
```suggestion
Description: 'An open-source Python module that offers an optimized tensor library for deep learning.'
```

File: content/python/concepts/modules/terms/pytorch/pytorch.md
Comment (yangc95) [inline content/python/concepts/modules/terms/pytorch/pytorch.md:17]:
```suggestion
**PyTorch** is a free [open-source](https://www.codecademy.com/resources/docs/general/open-source) Python module that offers an optimized tensor library for deep learning. It is based on the Torch library and can be used for applications such as natural language processing and computer vision. Released in 2016 and developed primarily by Meta (formerly Facebook) AI, deep learning software built on PyTorch include Tesla Autopilot and Uber's Pyro.
```

File: content/python/concepts/modules/terms/tqdm/tqdm.md
Comment (yangc95) [inline content/python/concepts/modules/terms/tqdm/tqdm.md:17]:
```suggestion
The **`tqdm`** module allows for the generation of progress bars in Python. The name is derived from the Arabic word "taqaddum", which translates as "progress." It is designed to have minimal overhead, using algorithms to predict the remaining time and to skip unnecessary iteration displays. It has about a 60ns (nanoseconds) overhead per iteration whereas other progress bar implementations can run an 800ns overhead per iteration. `tqdm` does not require any dependencies.
```

File: content/python/concepts/modules/terms/tqdm/tqdm.md
Comment (yangc95) [inline content/python/concepts/modules/terms/tqdm/tqdm.md:19]:
```suggestion
> **Note:** `tqdm` does require running in an environment that supports carriage return (`\r`) and line feed (`\n`) control characters.
```

**Review by yangc95 — COMMENTED** [review]
@SSwiniarski submitted a few suggestions 🙏

File: content/python/concepts/modules/modules.md
Comment (Dusch4593) [reply content/python/concepts/modules/modules.md:79]: I agree with this suggestion. And there actually is a place for both variations! If used as an adjective preceding a noun (as is the case, here), it should be "open-source". But if used more like a concept or noun (e.g., "Open source is...", "this is open source"), the hyphen should not be included.

**Comment by SSwiniarski** [comment]
@yangc95, I implemented your suggestions.

File: content/python/concepts/modules/terms/pyspark/pyspark.md
Comment (yangc95) [inline content/python/concepts/modules/terms/pyspark/pyspark.md:19]:
```suggestion
**PySpark** is a Python API for Apache Spark, consisting of several modules. Apache Spark (or Spark) is an analytical engine for large-scale distributed data processing and [machine learning](https://www.codecademy.com/resources/docs/general/machine-learning). Spark can run operations with very large datasets on distributed clusters about 100 times faster than traditional Python applications. PySpark has seen wide usage in the data science and machine learning communities due to the extensive number of data science libraries available in Python, and its ability to efficiently process large datasets.
```

File: content/python/concepts/modules/terms/pytorch/pytorch.md
Comment (yangc95) [inline content/python/concepts/modules/terms/pytorch/pytorch.md:17]: Read that it's a framework -- wanting to confirm

File: content/python/concepts/modules/terms/tqdm/tqdm.md
Comment (yangc95) [inline content/python/concepts/modules/terms/tqdm/tqdm.md:17]:
```suggestion
The **`tqdm`** module allows for the generation of progress bars in Python. The name is derived from the Arabic word, "taqaddum," which translates as "progress." It is designed to have minimal overhead, using algorithms to predict the remaining time and to skip unnecessary iteration displays. It has about a 60ns (nanoseconds) overhead per iteration whereas other progress bar implementations can run an 800ns overhead per iteration. `tqdm` does not require any dependencies.
```

File: content/python/concepts/modules/terms/tqdm/tqdm.md
Comment (yangc95) [inline content/python/concepts/modules/terms/tqdm/tqdm.md:19]:
```suggestion
> **Note:** `tqdm` requires an environment that supports carriage return (`\r`) and line feed (`\n`) control characters.
```

**Review by yangc95 — COMMENTED** [review]
@SSwiniarski thanks for making those changes! Upon a second look, I made a few more suggestions. I think these entries should be good to merge after

File: content/python/concepts/modules/terms/pytorch/pytorch.md
Comment (SSwiniarski) [reply content/python/concepts/modules/terms/pytorch/pytorch.md]: That seems correct. I edited the entry to reflect that.

**Comment by SSwiniarski** [comment]
@yangc95 I implemented your latest updates.

**Review by yangc95 — APPROVED** [review]

**Comment by SSwiniarski** [comment]
@Dusch4593, Does this look good to merge, or do you think we need another review?

**Comment by Dusch4593** [comment]
I don't think another review will be needed, but I made some edits to `modules.md` in 13a82f448219eb3965cb4c53817daefc68014b8e and 945526fc2f0ed9a58db5aae6e7b45adab7b23914 that include:

- Adding a link to [General: File Formats](https://www.codecademy.com/resources/docs/general/file-formats) near the top of the entry.
- Some word-smithing in the "Creating and Importing Modules" section.
- Replaced the `collections` example to stick with the fictional `video_player` module (plus, `collections` is linked further down in the entry anyways).
- Removed instances of "you"/"your".

**Comment by Dusch4593** [comment]
Also, in f41cab95d673d1fe4cc50a3ba897f98f3928e3bc:
- I added links for the [API](https://www.codecademy.com/resources/docs/general/api) and [data science](https://www.codecademy.com/resources/docs/general/data-science) entries under `pyspark.md`.
- I removed the link to open source in `pytourch.md` (since it would break after #935 is merged) and turned link text for NumPy entry to CamelCase.

**Pull request merged**
